framework/src/security/security_auth.c

```diff
@@ -109,15 +109,15 @@ security_error auth_get_certificate(security_handle hnd, const char *cert_name,
 	hal_data cert_out = {ctx->data1, ctx->dlen1, NULL, 0};
 	hal_result_e hres = HAL_SUCCESS;
 
-	SECAPI_CALL(sl_get_certificate(ctx->sl_hnd, cert_idx, &cert_out, &hres));
-	if (hres != HAL_SUCCESS) {
-		SECAPI_HAL_RETURN(hres);
+	if (strncmp(cert_name, "ss/", 3) == 0) {
+		SECAPI_CALL(sl_get_certificate(ctx->sl_hnd, cert_idx, &cert_out, &hres));
+		if (hres != HAL_SUCCESS) {
+			SECAPI_HAL_RETURN(hres);
+		}
+	} else {
+		SECAPI_CALL(sl_get_factory_cert(ctx->sl_hnd, cert_idx, &cert_out, &hres));
```

Comment:
So, in order to call sl_get_factory_cert, is it okay that no other conditions exist? It might be better to check of cert_name for factory data (such as "factory/x")

Reply:
It has been removed. 
And There is an opinion what it is not good that cert_name put as parameters. 
Anyway, in future, we should decide how to pass cert name or key index to hal.